### DISCUSSION TITLE
MinSOC value in Foxess EM configuration

### DISCUSSION THREAD

**Discussion by andyhawkes** [nathanmarlor/foxess_em#338] [General]
I have recently installed the Foxess EM integration and configured it using the FoxESS Cloud option (as I haven't had change to rig up a physical Modbus integration yet).

FoxESS recommend having a 5% difference between "Min SOC" and "MinSOC (on grid)" and suggest values of 10% and 15% respectively - see point 20 in https://www.fox-ess.com/wp-content/uploads/2023/07/FAQ-FOX-EN.pdf

I therefore set the "Min SOC" config value to 10%.

When I checked the FoxESS app today after successfully fully charging my batter overnight, I noticed that "Min SOC (on grid)" had been set to 10%.

Is this expected?

Does FoxESS EM set the "on grid" value, or both?
If both, could we please have a config option to enable us to have different values?

**Comment by nathanmarlor** [comment]
Andy, EM will set the on grid value dynamically during the off-peak period, it's how it prevents discharge past the calculated target SoC. It will set it back to whatever you entered in the config at the end of the off-peak period. As Fox Cloud only allows them to be set together, EM has to also bundle the non-grid setting and also uses the value entered in the config.

Put simply, expect the on-grid value to change frequently and for both values to be identical at the end of the off-peak period.

It wouldn't be a major change to add in a config value for each, but unfortunately I'm struggling to find time to dedicate to EM ATM (it's working too well for me...). I'd happily take a PR though.

**Reply by andyhawkes** [reply]
Python is not my forte, but I'll take a look if I can.

I've only recently moved to a flexible Octopus tariff and am still trying to get everything optimised - it's complicated at the moment by the occasional "power ups" that they are running in my area where you get periods of free electricity when there's too much renewable capacity in the grid (normally notified the evening before for a 2-4pm slot), so sometimes I'm topping up the battery then which messes with the algorithm a bit.
